sources/packages/backend/apps/workers/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable } from "@nestjs/common";
+import { Brackets, DataSource } from "typeorm";
+import {
+  RecordDataModelService,
+  MSFAANumber,
+  Student,
+  Application,
+  OfferingIntensity,
+} from "@sims/sims-db";
+import * as dayjs from "dayjs";
+import { MAX_MSFAA_VALID_DAYS } from "@sims/utilities";
+import { SequenceControlService } from "@sims/services";
+
+/**
+ * Service layer for MSFAA (Master Student Financial Aid Agreement)
+ * numbers generated for a student.
+ */
+@Injectable()
+export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
+  constructor(
+    dataSource: DataSource,
+    private readonly sequenceService: SequenceControlService,
+  ) {
+    super(dataSource.getRepository(MSFAANumber));
+  }
+
+  /**
+   * Creates a new MSFAA record with a new number for the specified student.
+   * @param studentId student to have a new MSFAA record created.
+   * @returns Created MSFAA record.
+   */
+  async createMSFAANumber(
+    studentId: number,
+    referenceApplicationId: number,
+    offeringIntensity: OfferingIntensity,
+  ): Promise<MSFAANumber> {
+    const newMSFAANumber = new MSFAANumber();
+    newMSFAANumber.msfaaNumber = await this.consumeNextSequence(
+      offeringIntensity,
+    );
+    newMSFAANumber.student = { id: studentId } as Student;
+    newMSFAANumber.referenceApplication = {
+      id: referenceApplicationId,
+    } as Application;
+    newMSFAANumber.offeringIntensity = offeringIntensity;
+    return this.repo.save(newMSFAANumber);
+  }
+
+  /**
+   * Generates the next number for an MSFAA.
+   * @returns number to be used for the next MSFAA.
```

Comment:
param comment missing

Reply:
return comment says `number to be used for the next MSFAA.` but promise is `string`

Reply:
The number is not the "typescript number type" but the type of the string content returned.
A developer looking at the code alongside the comment will easily understand it, IMO.

Reply:
oh..okay